src/main/java/com/aws/greengrass/clientdevices/auth/metrics/ClientDeviceAuthMetrics.java

```diff
@@ -5,70 +5,62 @@
 
 package com.aws.greengrass.clientdevices.auth.metrics;
 
-
 import com.aws.greengrass.telemetry.impl.Metric;
+import com.aws.greengrass.telemetry.impl.MetricFactory;
 import com.aws.greengrass.telemetry.models.TelemetryAggregation;
 import com.aws.greengrass.telemetry.models.TelemetryUnit;
 
+import java.time.Clock;
 import java.time.Instant;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.atomic.AtomicLong;
+import javax.inject.Inject;
 
 public class ClientDeviceAuthMetrics {
+    private final AtomicLong subscribeToCertificateUpdatesSuccess = new AtomicLong();
+    private final MetricFactory mf = new MetricFactory(NAMESPACE);
+    private final Clock clock;
+    private static final String NAMESPACE = "aws.greengrass.clientdevices.Auth";
+    public static final String METRIC_SUBSCRIBE_TO_CERTIFICATE_UPDATES_SUCCESS =
+            "SubscribeToCertificateUpdates.Success";
 
-    private final AtomicLong certSubscribeSuccess = new AtomicLong();
-    private final AtomicLong certSubscribeError = new AtomicLong();
-    private final AtomicLong invalidConfig = new AtomicLong();
-    private final AtomicLong certRotation = new AtomicLong();
+    /**
+     * Constructor for Client Device Auth Metrics.
+     *
+     * @param clock Clock
+     */
+    @Inject
+    public ClientDeviceAuthMetrics(Clock clock) {
+        this.clock = clock;
+    }
 
-    private static final String NAMESPACE = "ClientDeviceAuth";
+    /**
+     * Emit metrics using Metric Factory.
+     */
+    public void emitMetrics() {
```

Comment:
Next PR this emit needs to be called on a timer like we do in the Nucleus for the builtin telemetry producers